src/main/scala/scorex/core/network/peer/PeerDatabaseImpl.scala

```diff
@@ -10,13 +10,13 @@ import scala.collection.JavaConverters._
 
 
 //todo: persistence
-class PeerDatabaseImpl(settings: NetworkSettings,
-                       filename: Option[String],
-                       timeProvider: NetworkTimeProvider) extends PeerDatabase {
+class PeerDatabaseImpl(bindAddress: InetSocketAddress,
```

Comment:
I think this is a good change.

@kushti, in general, I think we should be passing only parameters that are really needed for each component, instead of passing a big Settings object containing all possible parameters around everywhere.  What do you think? Should we add a task like this to the architecture review?

Reply:
A disadvantage of doing that, though, is that the parameter list becomes longer. Before we had only `settings: NetworkSettings`; now we have `bindAddress: InetSocketAddress` and `declaredAddress: Option[InetSocketAddress]`. Not a big deal in this case, but could be inconvenient when the component needs several parameters.

A middle-ground solution would be to split `NetworkSettings` into smaller settings. For instance, `bindAddress` and `declaredAddress` could be part of a `SelfSettings` inside `NetworkSettings`, and then we would pass `SelfSettings` to `PeerDatabaseImpl`.

Reply:
Also, anyway, if we move the self detection logic outside `PeerDatabaseImpl`, then we wouldn't even need to pass `bindAddress` and `declaredAddress` to `PeerDatabaseImpl`.

Reply:
@ceilican finer granularity for settings sounds like a good choice. If more than 2 parameters to pass, it is better use *Settings I suppose. Two params are ok.